miniconf_derive/src/lib.rs

```diff
@@ -124,25 +119,18 @@ fn metadata_arm((i, struct_field): (usize, &StructField)) -> proc_macro2::TokenS
         quote! {
             #i => {
                 let mut meta = <#field_type>::metadata();
-
-                // Unconditionally account for separator since we add it
-                // even if elements that are deferred to (`Options`)
-                // may have no further hierarchy to add and remove the separator again.
-                meta.max_length += stringify!(#field_name).len() + 1;
+                meta.max_length += 1 + stringify!(#field_name).len();
                 meta.max_depth += 1;
-
                 meta
             }
         }
     } else {
         quote! {
             #i => {
                 let mut meta = miniconf::Metadata::default();
-
-                meta.max_length = stringify!(#field_name).len();
+                meta.max_length = 1 + stringify!(#field_name).len();
```

Comment:
Should outline where the `+1` is coming from. I think it's the prefix separator? But also, is separator guaranteed to be a single `char`?

Reply:
Yes. It's a single `char`.
But the length of a single char is not neccessarily 1.

Reply:
deferred to #152

Reply:
This should be as easy as calling `Self::Separator.len_utf8()` - don't know if we should really defer that?

Reply:
Ah, it's because the derive doesn't know anything about the separator, so we need to add the number of "Separators" into the metadata later, which we then use later to calculate if we have enough buffer etc. for a given serde impl